### PR TITLE
Supported WASI-NN streaming extension for NNRPC.

### PR DESCRIPTION
**Pull request by arvganesh**
Addresses #3319. This PR did the following for the following functions: `ComputeSingle`, `GetOutputSingle`, and `FiniSingle`.
- Defined new RPCs and messages for the NNRPC server.
- Client Side: added RPC call code path for `ComputeSingle`, `GetOutputSingle`, `FiniSingle`
- Server Side: updated backend to call rust bindings for the given functions.
- Tests: Added corresponding tests for the given functions using the OpenVINO-backend.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: test/plugins/wasi_nn/wasi_nn.cpp [hunk 1]
```diff
@@ -155,12 +155,30 @@ TEST(WasiNNTest, OpenVINOBackend) {
   EXPECT_TRUE(FuncInst->isHostFunction());
   auto &HostFuncGetOutput =
       dynamic_cast<WasmEdge::Host::WasiNNGetOutput &>(FuncInst->getHostFunc());
+  // Get the function "get_output_single".
```
Comment (hydai) [inline test/plugins/wasi_nn/wasi_nn.cpp:158]: I got confused. Why do you add these tests to the OpenVINO backend? The OpenVINO backend does not implement these LLM-related extension functions. That's why the unit tests will fail when trying to get the non-existing functions.

**Review by hydai — CHANGES_REQUESTED** [review]

File: plugins/wasi_nn/wasinnfunc.cpp [hunk 2]
```diff
@@ -462,6 +454,29 @@ Expect<WASINN::ErrNo> WasiNNGetOutputSingle::bodyImpl(
...
+#ifdef WASMEDGE_BUILD_WASI_NN_RPC
+  if (Env.NNRPCChannel != nullptr) {
+    auto Stub = wasi_ephemeral_nn::GraphExecutionContextResource::NewStub(
+        Env.NNRPCChannel);
+    grpc::ClientContext ClientContext;
+    wasi_ephemeral_nn::GetOutputRequest Req;
+    Req.set_resource_handle(Context);
+    Req.set_index(Index);
+    wasi_ephemeral_nn::GetOutputResult Res;
+    auto Status = Stub->GetOutput(&ClientContext, Req, &Res);
```
Comment (AkihiroSuda) [inline plugins/wasi_nn/wasinnfunc.cpp:466]:
```suggestion
    auto Status = Stub->GetOutputSingle(&ClientContext, Req, &Res);
```

**Comment by hydai** [comment]
Hi @arvganesh 
Are you going to finish this PR?

File: plugins/wasi_nn/wasinnfunc.cpp (hunk 2 above)
Comment (AkihiroSuda) [reply plugins/wasi_nn/wasinnfunc.cpp:466]: ping @arvganesh

**Comment by hydai** [comment]
Closing this due to inactive. Replace by #3506

**Pull request closed without merging**